### PR TITLE
[Merged by Bors] - refactor(RingTheory/OreLocalization): Ore localization of modules

### PR DESCRIPTION
**Pull request by erdOne**
To accommodate this change, right ore condition has been changed to left ore.
This enables us to redefine `Localization` and `LocalizedModule` with `OreLocalization` in the future.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by joelriou — APPROVED** [review]

**Comment by joelriou** [comment]
It looks good to me, but I would let someone more into the algebra part of mathlib have the last say.

maintainer merge

File: Mathlib/RingTheory/OreLocalization/Basic.lean [hunk 1]
```diff
@@ -84,91 +87,94 @@ variable (R S) [OreSet S]
...
+macro_rules | `($R[$S⁻¹]) => ``(OreLocalization $S $R)
 
 attribute [local instance] oreEqv
 
 variable {R S}
+variable {X} [MulAction R X]
 
-/-- The division in the ore localization `R[S⁻¹]`, as a fraction of an element of `R` and `S`. -/
-def oreDiv (r : R) (s : S) : R[S⁻¹] :=
+/-- The division in the ore localization `X[S⁻¹]`, as a fraction of an element of `X` and `S`. -/
```
Comment (jcommelin) [inline Mathlib/RingTheory/OreLocalization/Basic.lean:97]: Can we make the notation `X[S⁻¹]` work?

It also seems that many docstrings below need some updating.

Comment (erdOne) [reply Mathlib/RingTheory/OreLocalization/Basic.lean:97]: Turns out the notation already works. I just thought it didn't.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors merge

**Pull request closed without merging**